pkg/controller/daemon/controller.go

```diff
@@ -155,6 +166,7 @@ func NewDaemonSetsController(kubeClient client.Interface) *DaemonSetsController
 		},
 	)
 	dsc.syncHandler = dsc.syncDaemonSet
+	dsc.podStoreSynced = dsc.podController.HasSynced
```

Comment:
The reason I did this was so unittesting is easier. Please either add a similar unittest, or just invoke the function directly